src/pretix/multidomain/urlreverse.py

```diff
@@ -43,36 +43,41 @@
 from .models import KnownDomain
 
 
-def get_event_domain(event, fallback=False, return_info=False):
+def get_event_domain(event, fallback=False, return_mode=False):
     assert isinstance(event, Event)
     if not event.pk:
         # Can happen on the "event deleted" response
-        return (None, None) if return_info else None
-    suffix = ('_fallback' if fallback else '') + ('_info' if return_info else '')
+        return (None, None) if return_mode else None
+    suffix = ('_fallback' if fallback else '') + ('_mode' if return_mode else '')
     domain = getattr(event, '_cached_domain' + suffix, None) or event.cache.get('domain' + suffix)
     if domain is None:
         domain = None, None
-        if fallback:
+        if hasattr(event, 'alternative_domain_assignment'):
+            domain = event.alternative_domain_assignment.domain_id, KnownDomain.MODE_ORG_ALT_DOMAIN
+        elif fallback:
             domains = KnownDomain.objects.filter(
-                Q(event=event) | Q(organizer_id=event.organizer_id, event__isnull=True)
+                Q(event=event, mode=KnownDomain.MODE_EVENT_DOMAIN) |
+                Q(organizer_id=event.organizer_id, event__isnull=True, mode=KnownDomain.MODE_ORG_DOMAIN)
             )
             domains_event = [d for d in domains if d.event_id == event.pk]
             domains_org = [d for d in domains if not d.event_id]
             if domains_event:
-                domain = domains_event[0].domainname, "event"
+                domain = domains_event[0].domainname, KnownDomain.MODE_EVENT_DOMAIN
             elif domains_org:
-                domain = domains_org[0].domainname, "organizer"
+                domain = domains_org[0].domainname, KnownDomain.MODE_ORG_DOMAIN
         else:
-            domains = event.domains.all()
-            domain = domains[0].domainname if domains else None, "event"
+            try:
+                domain = event.domain.domainname, KnownDomain.MODE_EVENT_DOMAIN
+            except KnownDomain.DoesNotExist:
+                domain = None, None
         event.cache.set('domain' + suffix, domain or 'none')
         setattr(event, '_cached_domain' + suffix, domain or 'none')
     elif domain == 'none':
         setattr(event, '_cached_domain' + suffix, 'none')
         domain = None, None
     else:
         setattr(event, '_cached_domain' + suffix, domain)
-    return domain if return_info or not isinstance(domain, tuple) else domain[0]
+    return domain if return_mode or not isinstance(domain, tuple) else domain[0]
```

Comment:
the `or not isinstance(domain, tuple)` condition feels like it either never triggers or if it does, can be a source for a very weird bug, where get_event_domain might return a non-tuple value when a tuple is expected?

Reply:
Yup, that case can probably go, I don't know why it is there

Reply:
```suggestion
    return domain if return_mode else domain[0]
```